zerver/lib/actions.py

```diff
@@ -2737,6 +2763,28 @@ def our_person(p):
                         bot.update(event['bot'])
 
         elif event['type'] == 'stream':
+            if event['op'] == 'create':
+                for stream in event['streams']:
+                    stream_name = stream['name']
+                    realm = get_realm('zulip.com')
+                    new_stream, created = create_stream_if_needed(realm, stream_name, stream['invite_only'])
+                    if not stream['invite_only']:
+                        subscribers = get_subscriber_ids(new_stream)
+                        stream_data = {
+                            'description': new_stream.description,
+                            'stream_id': new_stream.id,
+                            'subscribers':  subscribers,
+                            'invite_only': new_stream.invite_only,
+                            'name': new_stream.name,
+                            'email_address': encode_email_address_helper(new_stream.name, new_stream.email_token)
+                        }
+                        # Add stream to never_subscribed (if not invite_only)
+                        state['never_subscribed'].append(stream_data)
+
+            if event['op'] == 'delete':
+                for stream in event['streams']:
+                    state['stream'] = [x for x in state['stream'] if not x == stream]
```

Comment:
For consistency with other code in this file, please use `s` instead of `x` as the placeholder variable here (and in the other list comprehensions in this diff)